fili-core/src/main/java/com/yahoo/bard/webservice/web/apirequest/DataApiRequestImpl.java

```diff
@@ -294,6 +291,73 @@ public DataApiRequestImpl(
     }
 
     /**
+     * All argument constructor, meant to be used for rewriting apiRequest.
+     *
+     * @param format  Format for the response
+     * @param paginationParameters  Pagination info
+     * @param uriInfo  The URI info
+     * @param builder  A response builder
+     * @param table  Logical table requested
+     * @param granularity  Granularity of the request
+     * @param dimensions  Grouping dimensions of the request
+     * @param perDimensionFields  Fields for each of the grouped dimensions
+     * @param logicalMetrics  Metrics requested
+     * @param intervals  Intervals requested
+     * @param apiFilters  Global filters
+     * @param havings  Top-level Having caluses for the request
+     * @param having  Single global Druid Having
+     * @param sorts  Sorting info for the request
+     * @param count  Global limit for the request
+     * @param topN  Count of per-bucket limit (TopN) for the request
+     * @param asyncAfter  How long in milliseconds the user is willing to wait for a synchronous response
+     * @param timeZone  TimeZone for the request
+     * @param filterBuilder  A builder to use when building filters for the request
+     * @param havingApiGenerator  A generator to generate havings map for the request
+     * @param dateTimeSort  A dateTime sort column with its direction
+     */
+    protected DataApiRequestImpl(
```

Comment:
This constructor already exists before this PR. The change I made was changing the method modifier from `private` to `protected` so that the testing class, which extends this `DataApiRequestImpl`, can construct testing `DataApiRequest`